MotorI2C/motorI2C.cpp

```diff
@@ -0,0 +1,57 @@
+#include <Wire.h>
+#include "MotorI2C.h"
+#include "SmartMotor.h"
+
+// Create a custom I2C instance for the Pico (as slave)
+TwoWire I2C_Slave = TwoWire(0);  // Use bus 0 for the I2C
+
+MotorI2C::MotorI2C(TwoWire& wire, int slaveAddress) 
+  : wireInstance(wire), slaveAddress(slaveAddress), receivedLeftSpeed(0), receivedRightSpeed(0), newDataAvailable(false), lastCommandTime(0) {}
+
+// Setup I2C communication
+void MotorI2C::begin() {
+  wireInstance.begin(slaveAddress);  // Initialize the I2C as a slave
+  wireInstance.onReceive(receiveData);  // Set the receive function to handle incoming data
+  wireInstance.onRequest(sendTelemetry); // Set the request function to handle telemetry data
```

Comment:
Since we have the motors here we can decide if we want to `SmartMotor#begin()` and `SmartMotor#calibrate()` here or outside the library, but at the moment these necessary functions are not called anywhere

Reply:
We are trying to fix all problems, in this case we understood what is the issue, but we don't know which solution is better. What do you suggest?

Reply:
You said that we have to decide where to call this functions (in the library or outside) but we don't know which option is better. We were asking you for a suggestion.

Reply:
Ops, I read a not which was not there 😅